### PR TITLE
Adding more programs for userland

### PR DESCRIPTION
**Pull request by elbing**
- cdfs: classic, maybe we need an alternate ata driver, I'm not sure if there are CD/DVD units
   AHCI
- map and plot: classic demo for graphics
- tarsplit: another utility for compressing
- zenith: Keith Poole's acme version focused on keyboard

Signed-off-by: Álvaro Jurado <elbingmiss@gmail.com>

### CODE REVIEW

**Comment by elbing** [comment]
Remaining for complete classic userland (discarding experiments or tests like cmux and syscall, or obsoletes like ssh1 and vt):
```
aquarela
cfs
cifs
dict
execnet
ext2srv
fax
grap
htmlroff
lp
paqfs
pic
postscript
proof
ratfs
scat
scuzz
snap
spell
spin
ssh1
ssh2
tapefs
tbl
tcs
troff2html
vac
vnc
webfs
wikifs
```
Maybe another few into cmd/ip or cmd/aux, not sure.

File: sys/src/cmd/map/libmap/simpleconic.c [hunk 1]
```diff
@@ -11,33 +11,33 @@
 #include <libc.h>
 #include "map.h"
 
-static double r0, a;
+static double rr0, a;
```
Comment (gmacd) [inline sys/src/cmd/map/libmap/simpleconic.c]: Out of interest, why was this renamed?  Did it clash with something?

File: sys/src/cmd/cdfs/main.c [hunk 2]
```diff
@@ -382,14 +383,14 @@ readctl(Req *r)
 	p = seprint(p, e, "maxspeed read %d write %d\n",
 		drive->maxreadspeed, drive->maxwritespeed);
 
-	if (drive->scsi.changetime != 0 && drive->ntrack != 0) { /* have disc? */
+	if (drive->changetime != 0 && drive->ntrack != 0) { /* have disc? */
```
Comment (gmacd) [inline sys/src/cmd/cdfs/main.c]: This looks like a bugfix - maybe would have been better to put this in separately?

Comment (elbing) [reply sys/src/cmd/cdfs/main.c]: Derivated from deanon (abandoning anonymous structs in Plan9) that was done later than the first attempt to port cdfs. If you have a better solution... tell me.

File: sys/src/cmd/cdfs/main.c [hunk 3]
```diff
@@ -350,7 +351,7 @@ static void
 readctl(Req *r)
 {
 	int i, isaudio;
-	uint32_t nwa;
+	uint64_t nwa;
```
[the remark this answers is not in the document]
Comment (elbing) [reply sys/src/cmd/cdfs/main.c]: Below is wrongly assigned to an unsigned integer. clang said. It's nwa = ~0ul I think, which I casted to (uint64_t). Now I see it, the cast should be uint32_t. I'll fix it.

File: sys/src/cmd/cdfs/main.c (hunk 2 above)
Comment (gmacd) [reply sys/src/cmd/cdfs/main.c]: In this case, scsi was already a field in the Drive struct - it wasn't anonymous.  Both the Scsi and Drive structs have changetime fields.

File: sys/src/cmd/cdfs/main.c [hunk 4]
```diff
@@ -382,14 +383,14 @@ readctl(Req *r)
...
-	if (drive->scsi.changetime != 0 && drive->ntrack != 0) { /* have disc? */
+	if (drive->changetime != 0 && drive->ntrack != 0) { /* have disc? */
 		ty = disctype(drive);
 		p = seprint(p, e, "%s", ty);
 		free(ty);
 		if (drive->mmctype != Mmcnone) {
 			nwa = getnwa(drive);
 			p = seprint(p, e, " next writable sector ");
-			if (nwa == ~0ul)
+			if (nwa == (uint64_t)~0ul)
```
Comment (elbing) [inline sys/src/cmd/cdfs/main.c]: it should be uint32_t, see comment above.

File: sys/src/cmd/map/libmap/simpleconic.c (hunk 1 above)
Comment (elbing) [reply sys/src/cmd/map/libmap/simpleconic.c]: In libc.h r0 it's a function, since it's redeclared here, compiler didn't like. Simpler solution was renaming it.

File: sys/src/cmd/tarsplit/tar.h [hunk 5]
```diff
@@ -48,19 +48,19 @@ typedef struct {
 } Header;
 
 typedef union {
-	uchar	dummy[Tblock];
-	Header;
+	uint8_t	dummy[Tblock];
+	Header *header;
```
[the remark this answers is not in the document]
Comment (elbing) [reply sys/src/cmd/tarsplit/tar.h]: I'm really not sure. Probably a value. I'll fix it.

File: sys/src/cmd/zenith/edit.c [hunk 6]
```diff
@@ -25,40 +25,40 @@ static char	linex[]="\n";
...
-	'<',	0,	0,	0,	0,	aDot,	0,	linex,	pipe_cmd,
-	'|',	0,	0,	0,	0,	aDot,	0,	linex,	pipe_cmd,
-	'>',	0,	0,	0,	0,	aDot,	0,	linex,	pipe_cmd,
+	{'\n',	0,	0,	0,	0,	aDot,	0,	0,	nl_cmd},
+	{'a',	1,	0,	0,	0,	aDot,	0,	0,	a_cmd},
+	{'b',	0,	0,	0,	0,	aNo,	0,	linex,	b_cmd},
+	{'c',	1,	0,	0,	0,	aDot,	0,	0,	c_cmd},
+	{'d',	0,	0,	0,	0,	aDot,	0,	0,	d_cmd},
+	{'e',	0,	0,	0,	0,	aNo,	0,	wordx,	e_cmd},
+	{'g',	0,	1,	0,	'p',	aDot,	0,	0,	g_cmd},
```
[the remark this answers is not in the document]
Comment (elbing) [reply sys/src/cmd/zenith/edit.c]: The editor, I guess. I'll fix it.

File: sys/src/cmd/cdfs/main.c (hunk 2 above)
Comment (elbing) [reply sys/src/cmd/cdfs/main.c]: compiler warned. Let me back to original an see what's happening.
